lorentz.py

```diff
@@ -148,26 +151,37 @@ def __init__(self, pairwise_matrix, sample_size=10):
         self.pairwise_matrix = pairwise_matrix
         self.n_items = len(pairwise_matrix)
         self.sample_size = sample_size
+        self.arange = np.arange(0, self.n_items)
 
     def __len__(self):
         return self.n_items
 
     def __getitem__(self, i):
-        I = torch.Tensor([i]).squeeze().long()
-        while True:
-            j = random.randint(0, self.n_items - 1)
-            if j != i:
+        I = torch.Tensor([i + 1]).squeeze().long()
+        has_child = False
+        arange = np.random.permutation(self.arange)
+        for j in arange:
```

Comment:
pairwise matrix is numpy no? `has_child = (self.pairwise_matrix[i] > 0 ).sum() > 0` will be faster I think?

Reply:
yes, changing